src/server.js

```diff
@@ -1,34 +1,75 @@
 const bodyParser = require('body-parser');
 const express = require('express');
 const session = require('express-session');
-
+const bcrypt = require('bcrypt');
+const User = require('./user');
 const STATUS_USER_ERROR = 422;
 const BCRYPT_COST = 11;
-
 const server = express();
-// to enable parsing of json bodies for post requests
 server.use(bodyParser.json());
-server.use(session({
-  secret: 'e5SPiqsEtjexkTj3Xqovsjzq8ovjfgVDFMfUzSmJO21dtXs4re'
-}));
+server.use(
+    session({
+        secret: 'e5SPiqsEtjexkTj3Xqovsjzq8ovjfgVDFMfUzSmJO21dtXs4re'
+    })
+);
 
-/* Sends the given err, a string or an object, to the client. Sets the status
- * code appropriately. */
-const sendUserError = (err, res) => {
-  res.status(STATUS_USER_ERROR);
-  if (err && err.message) {
-    res.json({ message: err.message, stack: err.stack });
-  } else {
-    res.json({ error: err });
-  }
-};
+server.post('/users', (req, res) => {
+    const { username, password } = req.body;
 
-// TODO: implement routes
+    if (username && password) {
+        bcrypt.hash(password, BCRYPT_COST, (err, hash) => {
+            if (err) {
+                res.status(STATUS_USER_ERROR).json(err);
+            }
+            const user = new User({ username, passHash: hash });
+            user.save()
+                .then((item) => {
+                    res.status(200).json(item);
+                })
+                .catch((error) => {
+                    res.status(STATUS_USER_ERROR).json(err);
```

Comment:
`res.json(err)` doesn't actually send the entire Error Object to the client. You can do something like `res.json({ message: err.message })`

The Error Object has a `message` property on it. Read more about the Error Object: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error